lib/fauxpaas/release.rb

```diff
@@ -7,32 +7,29 @@ module Fauxpaas
   class Release
 
     # @param deploy_config [DeployConfig]
-    # @param source_path [Pathname]
-    # @param shared_path [Pathname]
-    # @param unshared_path [Pathname]
-    def initialize(deploy_config:, source_path:, shared_path:, unshared_path:)
-      @deploy_config = deploy_config
-      @source_path = source_path
-      @shared_path = shared_path
-      @unshared_path = unshared_path
+    # @param artifact [Artifact]
+    def initialize(signature:, fs:,
+      artifact_factory: Artifact,
+      deploy_config_factory: DeployConfig)
+      @artifact = artifact_factory.new(signature: signature, fs: fs)
```

Comment:
The only real reason to do this here is so that we can more easily inject a constructor here rather than having to stub behavior on `Artifact` or `DeployConfig` in the specs. If you prefer, we could remove the word `factory` and replace it with `class`.

Reply:
In order of preference, I like best: factory + for, factory + new, class + new. I'm not terribly offended by the contract for a given factory being `it responds_to?(:new)`. No objection to leaving this one as is.